src/main/java/com/zendesk/maxwell/schema/MysqlPositionStore.java

```diff
@@ -156,24 +156,37 @@ public Long getLastHeartbeatSent() {
 		return lastHeartbeat;
 	}
 
+	private Position positionFromResultSet(ResultSet rs) throws SQLException {
+		if ( !rs.next() )
+			return null;
+
+		String gtid = gtidMode ? rs.getString("gtid_set") : null;
+		BinlogPosition pos = new BinlogPosition(
+			gtid,
+			null,
+			rs.getLong("binlog_position"),
+			rs.getString("binlog_file")
+		);
+
+		return new Position(pos, rs.getLong("last_heartbeat_read"));
+	}
+
+	public Position getAny() throws SQLException {
```

Comment:
Would it be worth instead having `getLatest()` (sorted by `last_heartbeat_read`)? That way you don't pick up an offset from an old abandoned client.

Also, should this filter `gtid_set=<am-i-using-gtid>` to get only a compatible position?

Reply:
yup, good point on sorting, will change. 

re: GTID... hmmm.  If we have an old client running non-gtid, and we boot a new client with gtid, we still want to borrow the binlog position of the old client, or else we'll run into the same issue with schema sharing.  But I don't think it'd work properly to run in GTID mode with that old, non-gtid position.  

For that matter, Maxwell has *never* had a decent story around a graceful non-GTID -> GTID upgrade path; I think our answer has always been "drop database maxwell;".  I mean try reading the mysql docs on how to get an initial GTID lock, it's really quite bad.

Reply:
Fair point, I don't know how a transition would actually work.